compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -413,12 +419,13 @@ impl<'mir, 'tcx> Checker<'mir, 'tcx> {
         }));
 
         let errors = ocx.select_all_or_error();
-        if !errors.is_empty() {
+        if errors.is_empty() {
+            Some(ConstConditionsHold::Yes)
+        } else {
             tcx.dcx()
                 .span_delayed_bug(call_span, "this should have reported a ~const error in HIR");
+            Some(ConstConditionsHold::No)
```

Comment:
change this to return the `ERrorGuaranteed` to make it clear that we can avoid the check as we've already errored?

Reply:
or hmm, why do we not emit a conditionally const error in this branch :thinking: I feel like we should either not report an error at all or talk about the const-condition which doesn't hold

Reply:
bc we don't check effect predicates hold in HIR if the const_trait_impl feature gate is not enabled, so we need to report an error in the const checker.

Reply:
Actually, I have no idea what you're asking for here. In the situation that you call a function with unsatisfied const conditions:

1. The `const_trait_impl` feature gate is enabled, then we report an error in HIR. This exists only as a sanity check so we detect cases where `~const` predicates are not checked in the type checker.
2. The gate is not enabled, in which case we report a `ConditionallyConstCall` error later on, which is gated on the `const_trait_impl` feature gate.

In the latter case, we don't want to mention what `~const` predicate does not hold if we don't even have the `const_trait_impl` feature gate enabled. Users of stable rust should not be concerned with const predicates.